components/CollectivePickerAsync.js

```diff
@@ -79,7 +79,7 @@ const CollectivePickerAsync = ({ types, limit, hostCollectiveIds, preload, filte
   const [searchCollectives, { loading, data }] = useLazyQuery(searchQuery);
   const [term, setTerm] = React.useState(null);
   const { formatMessage } = useIntl();
-  const collectives = (data && data.search && data.search.collectives) || [];
+  const collectives = term === '' ? [] : (data && data.search && data.search.collectives) || [];
```

Comment:
**[:no_entry_sign: **Blocker**]** This introduces a regression in the host dashboard, in there the collectives are preloaded even when there's no search on purpose, because they are already filtered through the host id.

I can suggest the following:

```suggestion
  const collectives = ((term || preload) && data?.search?.collectives) || [];
```


---

<details>
<summary>More details about this feedback</summary>
<br/>

This feedback categorization is inspired by [Netlify's Feedback Ladders system](https://www.netlify.com/blog/2020/03/05/feedback-ladders-how-we-encode-code-reviews-at-netlify). It includes 5 levels of feedback:

:rotating_light: **Critical blocker**
This feedback blocks all related work, and requires immediate action. 

:no_entry_sign: **Blocker**
This feedback should block the work from moving forward or being approved, but it does not necessarily require immediate action from the team.

:hammer_and_wrench: **Important**
This feedback should not block the work from moving forward or being approved, but it does require future action from the team. This generally means the feedback is not required for the MVP or initial draft of this work, but should be addressed at some point and we should ideally create an issue to track it.

:sparkles: **Enhancement**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they agree with it.

:ice_cream: **Details & personal preferences**
This feedback should not block the work from moving forward or being approved. The project lead or owner should consider the feedback and implement or open an issue it if they deem it useful. This is also the category for everything that relates to personal preferences or code style.

</details>